### PR TITLE
Visibility of state variable section added

### PR DESCRIPTION
**Pull request by shikhar360**
#### Added one section in the "More on variables and types" content slide drawer , 
- The section defines the types of visibility of state variables 
- What should be the appropriate visibility that can be given to a variable

### New Features *(required)*

Fixes # 
Understanding about the visibility of variables.
Connects #
null

### CODE REVIEW

File: lessons/projects/1.mdx [hunk 1]
```diff
@@ -710,4 +722,4 @@ Can a view function modify the state of the blockchain?
 What can we use a smart contract for?
 
 **Now, go to the community in Discord to share your new Open Sourcerer
-powers**......and find the answers too!!! 
+powers**......and find the answers too!!!
```
Comment (7i7o) [inline lessons/projects/1.mdx:725]: For future reference, check if you modified lines by accident 😊 
In this case it's really not important, but in big PRs, it creates a little more unnecessary work for the reviewer.

**Review by 7i7o — CHANGES_REQUESTED** [review]
Thanks a lot for your work @shikhar360 !!!

We are really happy that you managed to contribute and help us build this platform!

I'm adding some comments for you to change so we can merge your PR

Thanks a lot!!! 😊

**Comment by shikhar360** [comment]
I have made the requested changes @7i7o 😊 . Hope its good to go.

**Comment by PBillingsby** [comment]
@7i7o is this good to go now?

File: lessons/projects/1.mdx [hunk 2]
```diff
@@ -284,6 +284,24 @@ contract WAGMI {
         - **Struct:** Solidity allows users to create and define their own type in the form of structures. The structure is a group of different types even though it’s not possible to contain a member of its own type. The structure is a reference type variable which can contain both value types and reference types.
     - **Mapping Type**: Stores the data in a key-value pair where a key can be any value types. It is like a hash table or dictionary as in any other programming language, where data can be retrieved by key.
 
+    ### Visibility of State Variables
+
+    In Solidity , there are 3 types of variable visibility modifiers that a developer can use :
+
+    - **Public**: If a state variable visibility is marked as `public` then those variables can be used within the smart contract and can be accessed by other contracts as well.Compilers automatically generates functions to read its value when calling the contract, they are called 'getter' functions
```
Comment (7i7o) [inline lessons/projects/1.mdx:291]: > ... as well.Compilers automatically generates functions ...

... as well. Compilers automatically generate functions ...

**Review by 7i7o — CHANGES_REQUESTED** [review]
Hey @shikhar360 !
Thank you for your swift corrections, and sorry for being so busy to review this.

I added some **super minimal** language corrections, and it should be good to go

Ping me whenever you do this so we can merge ASAP 😊 

Also, thank you @PBillingsby for the ping 🙏

**Comment by shikhar360** [comment]
@7i7o Fixed the typos 😊 . You now can review this again

**Review by 7i7o — APPROVED** [review]
Everything is good to go!

**Pull request merged**